go/vt/vttablet/tabletmanager/vreplication/vdbclient.go

```diff
@@ -56,16 +56,19 @@ func (vc *vdbClient) Begin() error {
 	if vc.InTransaction {
 		return nil
 	}
-	if err := vc.DBClient.Begin(); err != nil {
-		return err
+	if vc.maxBatchSize > 0 {
+		// We are batching the contents of the transaction, which
+		// starts with the BEGIN and ends with the COMMIT, so we
+		// do not send a BEGIN down the wire ahead of time.
+		vc.queriesPos = int64(len(vc.queries))
+		vc.batchSize = 6 // begin and semicolon
+	} else {
+		// We're not batching so we start the transaction here
+		// by sending the BEGIN down the wire.
+		if err := vc.DBClient.Begin(); err != nil {
+			return err
+		}
```

Comment:
This is the fix for the noted extraneous BEGIN that surfaced in the unit tests. This was harmless, but also unnecessary and corrected here.